src/model/helpers.rs

```diff
@@ -903,27 +903,61 @@ pub struct GetNotificationsParams {
     // the notification ID to start returning messages after. Null to start with the most recent notification
     #[serde(rename = "aft")]
     pub after: Option<Uuid>,
+
+    // order unread notifications before read notifications regardless of time and paginate from there
+    #[serde(rename = "urf", default)]
+    pub unread_first: bool,
 }
 
 #[instrument(skip(postgres, metrics))]
 pub async fn get_notifications_for_subscriber(
     subscriber: Uuid,
-    GetNotificationsParams { limit, after }: GetNotificationsParams,
+    GetNotificationsParams {
+        limit,
+        after,
+        unread_first,
+    }: GetNotificationsParams,
     postgres: &PgPool,
     metrics: Option<&Metrics>,
 ) -> Result<GetNotificationsResult, sqlx::error::Error> {
-    let after_clause = if after.is_some() {
+    let after_clause = |after: Option<Uuid>, variable: &str| {
+        if after.is_some() {
+            let (unread_first_clause_left, unread_first_clause_right) = if unread_first {
+                (
+                    format!("(SELECT is_read FROM subscriber_notification WHERE id={variable}),"),
+                    "subscriber_notification.is_read,",
+                )
+            } else {
+                ("".to_owned(), "")
+            };
+            format!(
+                "
+                AND (
+                    {unread_first_clause_left}
+                    NOW() - (SELECT created_at FROM subscriber_notification WHERE id={variable}),
```

Comment:
Looks like there's no way to reverse the comparison for a specific field in the tuple, so have to make it subtract from a shared value to make it sort in reverse.